grakn-core/src/main/java/ai/grakn/engine/Jacksonisable.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.engine;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+
+/**
+ * <p>
+ *     Jacksonisable Object
+ * </p>
+ *
+ * <p>
+ *     Use to pass the necessary annotation to all objects which we use Jackson for serliasation and deserilisation
+ * </p>
+ *
+ * @author Filipe Peliz Pinto Teixeira
+ */
+@JsonIgnoreProperties(value={ "@id", "roles", "types", "graql", "rules" }, allowGetters=true)
+public interface Jacksonisable {
```

Comment:
Can we move `@id` into this interface?

Reply:
This will lead to a fun generic but I will give it a try.

Reply:
Actually scratch that. I am not sure how to do it because they all use it in different contexts. For example `ai.grakn.Keyspace` uses it but does not have a `@id` json property. It IS the `@id` property.

On the other hand, `ai.grakn.engine.controller.response.Concept` and `ai.grakn.engine.controller.response.Keyspace` use it and contain `@id` as a json property.

All 3 require the annotation provided by `Jacksonisable`. In other words if you want `@id` to be common place then another intr=erface might be needed and I am not sure we will gain much from that.